ddtrace/appsec/_api_security/api_manager.py

```diff
@@ -0,0 +1,173 @@
+            log.info("%s already enabled", cls.__name__)
+            return
+
+        log.info("Enabling %s", cls.__name__)
+        metrics.enable()
+        cls._instance = cls()
+        cls._instance.start()
+        log.info("%s enabled", cls.__name__)
+
+    @classmethod
+    def disable(cls):
+        # type: () -> None
+        if cls._instance is None:
+            log.info("%s not enabled", cls.__name__)
+            return
+
+        log.info("Disabling %s", cls.__name__)
+        cls._instance.stop()
+        cls._instance = None
+        metrics.disable()
+        log.info("%s disabled", cls.__name__)
+
+    def __init__(self):
+        # type: () -> None
+        super(APIManager, self).__init__()
+        try:
+            self.INTERVAL_PER_ROUTE = float(os.environ.get("_DD_API_SECURITY_INTERVAL_PER_ROUTE", "15.0"))
+        except BaseException:
+            pass
+
+        self._schema_meter = metrics.get_meter("schema")
+        self._log_limiter = RateLimiter(limit_rate=1.0, raise_on_exceed=False)
+        self._global_rate_limiter = RateLimiter(
+            limit_rate=self.GLOBAL_RATE_LIMIT,
+            raise_on_exceed=False,
+        )
+        self._rate_limiter_by_route = dict()  # type: Dict[str, RateLimiter]
+
+        log.info("%s initialized", self.__class__.__name__)
+
+    def _stop_service(self):
+        # type: () -> None
+        remove_context_callback(self._schema_callback, global_callback=True)
+
+    def _start_service(self):
+        # type: () -> None
+        add_context_callback(self._schema_callback, global_callback=True)
+
+    def _should_collect_schema(self, env):
+        if self.INTERVAL_PER_ROUTE <= 0.0:
+            return True
+        method = env.waf_addresses.get(SPAN_DATA_NAMES.REQUEST_METHOD)
+        route = env.waf_addresses.get(SPAN_DATA_NAMES.REQUEST_ROUTE)
+        # Framework is not fully supported
+        if not method or not route:
+            log.debug("unsupported groupkey for api security [method %s] [route %s]", bool(method), bool(route))
+            return False
+        # WAF has triggered, always collect schemas
+        results = env.telemetry.get(_WAF_RESULTS)
+        if results and any((result.data for result in results[0])):
+            return True
+        # Rate limit per route
+        key = method + route
+        r = self._rate_limiter_by_route.get(key)
+        if r is None:
+            self._rate_limiter_by_route[key] = r = RateLimiter(
+                limit_rate=1 / self.INTERVAL_PER_ROUTE,
+                tau=self.INTERVAL_PER_ROUTE,
+                raise_on_exceed=False,
+            )
+        return r.limit() is not RateLimitExceeded
+
+    def _schema_callback(self, env):
+        if env.span is None:
+            return
+        root = env.span._local_root or env.span
+        if not root or any(meta_name in root._meta for _, meta_name, _ in self.COLLECTED):
+            return
+
+        root._metrics["_dd.appsec.api_security.enabled"] = 1.0
```

Comment:
can we move "_dd.appsec.api_security.enabled" to constants?

Reply:
yes !